test/documentation_generator/generate_markdown_test.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:args/command_runner.dart';
+import 'package:cli_tools/cli_tools.dart';
+import 'package:test/test.dart';
+
+class AddSpiceCommand extends Command {
+  @override
+  final String name = 'add';
+
+  @override
+  final String description = 'Add something to the spice mix';
+
+  AddSpiceCommand() {
+    argParser.addOption('curry', help: 'Include curry in the spice mix.');
+    argParser.addOption('pepper', help: 'Include pepper in the spice mix.');
+  }
+
+  @override
+  void run() {}
+}
+
+class RemoveSpiceCommand extends Command {
+  @override
+  final String name = 'remove';
+
+  @override
+  final String description = 'Remove something from the spice mix';
+
+  RemoveSpiceCommand() {
+    argParser.addOption('curry', help: 'Remove curry from the spice mix.');
+    argParser.addOption('pepper', help: 'Remove pepper from the spice mix.');
+  }
+
+  @override
+  void run() {}
+}
+
+class AddVegetableCommand extends Command {
+  @override
+  final String name = 'add';
+
+  @override
+  final String description = 'Add a vegetable to your dish.';
+
+  AddVegetableCommand() {
+    argParser.addOption('carrot', help: 'Adds a fresh carrot to the dish.');
+  }
+
+  @override
+  void run() {}
+}
+
+class SpiceCommand extends BetterCommand {
+  @override
+  final String name = 'spice';
+
+  @override
+  final String description = 'Modifies the spice mix in your dish.';
+
+  SpiceCommand() {
+    addSubcommand(AddSpiceCommand());
+    addSubcommand(RemoveSpiceCommand());
+  }
+
+  @override
+  void run() {}
+}
+
+class VegetableCommand extends BetterCommand {
+  @override
+  final String name = 'vegetable';
+
+  @override
+  final String description = 'Add or remove vegatables to your dish.';
+
+  VegetableCommand() {
+    addSubcommand(AddVegetableCommand());
+  }
+
+  @override
+  void run() {}
+}
+
+void main() {
```

Comment:
Question: Should we add a test that validates that multiple commands are supported and that each command ends up in its own file?

Reply:
I have this test already, is it enough? Or did you mean we should also verify the content in the other file?
```dart
 test('then outputs each command into a separate file', () {
      expect(output.keys, containsAll(['vegetable.md', 'spice.md']));
    });
   ```

Reply:
Aaah, that is probably enough! I missed that! 👍